torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1715,6 +1715,66 @@ std::vector<For*> LoopNest::reorder(
   return result;
 }
 
+void LoopNest::tile(For* x, For* y, int x_factor, int y_factor) {
+  auto parent = dynamic_cast<Block*>(x->get_parent());
+  if (parent == nullptr) {
+    throw malformed_input("parent of the loops must be a Block");
+  }
+  if ((x->body()->nstmts() != 1) || (x->body()->front() != y)) {
+    throw malformed_input("two loops must be perfectly nested");
+  }
+
+  // Split x, y axes by x_factor and y_factor
+  For *yo, *yi;
+  splitWithMask(y, y_factor, &yo, &yi);
+  For *xo, *xi;
+  splitWithMask(x, x_factor, &xo, &xi);
+
+  // Re-identify yo, yi
+  bool handle_conds = false;
+  ExprHandle conds;
+  if (auto* xcond = dynamic_cast<Cond*>(xi->body()->front())) {
+    conds = ExprHandle(xcond->condition());
+    handle_conds = true;
+    yo = dynamic_cast<For*>(xcond->true_stmt()->front());
+  } else {
+    yo = dynamic_cast<For*>(xi->body()->front());
+  }
+  CHECK(yo);
+  yi = dynamic_cast<For*>(yo->body()->front());
+  CHECK(yi);
```

Comment:
Is this needed because `splitWithMask` clones the statements in the body of the `For`?

Reply:
Yeah, `splitWithMask` clones the statements in the body : https://github.com/pytorch/pytorch/blob/b5b158a6c6de94dfb983b447fa33fea062358844/torch/csrc/jit/tensorexpr/loopnest.cpp#L1237
Can we do `Substitute` on the original stmt? If so, what's the main reason for cloning?

Reply:
Cloning is probably a leftover from some old implementation that can be removed now.

Reply:
Yeah, this cloning part should be removed.